phoenix-core/src/test/java/org/apache/phoenix/monitoring/MetricUtilTest.java

```diff
@@ -48,4 +52,17 @@ public void testGetMetricsStopWatchWithMetricsFalse() throws Exception {
                 LogLevel.OFF, WALL_CLOCK_TIME_MS);
         assertFalse(metricsStopWatch.getMetricsEnabled());
     }
+
+    @Test
+    //Check that MetricsSystemImpl has a String "prefix" field in the Hadoop version we test with
+    public void testInternalMetricsField() throws NoSuchFieldException,
+            SecurityException, IllegalArgumentException, IllegalAccessException {
+        MetricsSystemImpl metrics = (MetricsSystemImpl) DefaultMetricsSystem.instance();
+        Field prefixField = MetricsSystemImpl.class.getDeclaredField("prefix");
+        prefixField.setAccessible(true);
+        String oldValue = (String)prefixField.get(metrics);
+        prefixField.set(metrics, "dummy");
+        prefixField.set(metrics, oldValue);
+        prefixField.setAccessible(false);
```

Comment:
I was initially just thinking to make this `MetricUtil.isDefaultMetricsInitialized()` to avoid keeping this unit test aligned with the code in that method. I guess the likelihood of these drifting is low :)

Reply:
isDefaultMetricsInitialized() already has a catch-all section to avoid failing if the API changes, so it cannot really be tested.